apiserver/authentication/agent.go

```diff
@@ -24,11 +64,20 @@ type taggedAuthenticator interface {
 
 // Authenticate authenticates the provided entity.
 // It takes an entityfinder and the tag used to find the entity that requires authentication.
-func (*AgentAuthenticator) Authenticate(ctx context.Context, entityFinder EntityFinder, authParams AuthParams) (state.Entity, error) {
-	entity, err := entityFinder.FindEntity(authParams.AuthTag)
+func (a agentAuthenticator) Authenticate(ctx context.Context, authParams AuthParams) (state.Entity, error) {
```

Comment:
Seems odd to mix pointer/value receivers.

Reply:
This never modified the heap allocation, so we can just use the stack.